packages/ketcher-core/src/domain/entities/Phosphate.ts

```diff
@@ -94,20 +93,31 @@ export class Phosphate extends BaseMonomer {
     return PhosphateSubChain;
   }
 
-  get isPartOfRna() {
-    const previousMonomer =
-      this.attachmentPointsToBonds.R1?.getAnotherMonomer(this);
-    const isPreviousMonomerSugar = previousMonomer instanceof Sugar;
-    const isSugarConnectedToBase =
-      previousMonomer?.attachmentPointsToBonds.R3?.getAnotherMonomer(
-        previousMonomer,
-      ) instanceof RNABase;
-    const nextMonomer =
-      this.attachmentPointsToBonds.R2?.getAnotherMonomer(this);
-    const isNextMonomerRna = nextMonomer?.isPartOfRna;
+  public get isPartOfRna(): boolean {
+    // To avoid endless looping when checking `monomerForR2`,
+    // we take into account that we did not return to checking the initial phosphate (`this`).
+    const checkIfPhosphateIsPartOfRNA = (phosphate: Phosphate): boolean => {
+      const { R1: polymerBond1, R2: polymerBond2 } =
+        phosphate.attachmentPointsToBonds;
+      const monomerForR1 = polymerBond1?.getAnotherMonomer(phosphate);
+      const monomerForR2 = polymerBond2?.getAnotherMonomer(phosphate);
+      if (!monomerForR1 || !monomerForR2) {
+        return false;
+      }
+      const isMonomerForR1SugarAndPartOfRNA =
+        monomerForR1 instanceof Sugar && (monomerForR1 as Sugar).isPartOfRna;
```

Comment:
I believe that we don't need cast to Sugar type here because we already checked that it is instance of Sugar